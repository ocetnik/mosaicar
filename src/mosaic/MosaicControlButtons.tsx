import * as React from 'react';
import Share from './Share';

interface IMosaicControlButtonsProps {
    canvasImageLoaded: boolean;
    mosaicSvgString: string;
    onConvertButtonClick: () => Promise<void>
    onShareButtonClick: () => Promise<void>
}

class MosaicControlButtons extends React.Component<IMosaicControlButtonsProps, {}> {
    public render() {
        if (!this.props.canvasImageLoaded) {
            return <div>Loading original image</div>;
        }

        return this.props.mosaicSvgString === ''
            ? <div>
                <button onClick={this.props.onConvertButtonClick}>
                    Convert image into a mosaic
                </button>
            </div>
            : <Share onShareButtonClick={this.props.onShareButtonClick} />;
    }
}

export default MosaicControlButtons;
