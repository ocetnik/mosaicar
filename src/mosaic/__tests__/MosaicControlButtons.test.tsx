import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MosaicControlButtons from '../MosaicControlButtons';

describe('MosaicControlButtons', () => {
    const onConvertButtonClick = jest.fn();
    const onShareButtonClick = jest.fn();

    test('Show loading when canvasImageLoaded === false', () => {
        const component = renderer.create(
            <MosaicControlButtons
                canvasImageLoaded={false}
                mosaicSvgString={''}
                onConvertButtonClick={onConvertButtonClick}
                onShareButtonClick={onShareButtonClick}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Show Convert button when canvasImageLoaded === true && mosaicSvgString === \'\'', () => {
        const component = renderer.create(
            <MosaicControlButtons
                canvasImageLoaded={true}
                mosaicSvgString={''}
                onConvertButtonClick={onConvertButtonClick}
                onShareButtonClick={onShareButtonClick}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
