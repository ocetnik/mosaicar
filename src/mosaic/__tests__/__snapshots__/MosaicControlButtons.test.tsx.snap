// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MosaicControlButtons Show Convert button when canvasImageLoaded === true && mosaicSvgString === '' 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Convert image into a mosaic
  </button>
</div>
`;

exports[`MosaicControlButtons Show loading when canvasImageLoaded === false 1`] = `
<div>
  Loading original image
</div>
`;
